src/blueprint/operation.rs

```diff
@@ -0,0 +1,75 @@
+use async_graphql::dynamic::Schema;
+use async_graphql::ValidationMode;
+use derive_setters::Setters;
+
+use super::into_schema::{create, SchemaModifiers};
+use super::Blueprint;
+use crate::valid::{Cause, Valid, ValidationError};
+
+#[derive(Debug, Setters)]
+pub struct Operation {
+  query: String,
+  trace: Option<String>,
+}
+
+pub fn validation_schema(blueprint: &Blueprint) -> Result<Schema, ValidationError<String>> {
+  match create(blueprint, Some(SchemaModifiers { no_resolver: true }))
+    .validation_mode(ValidationMode::Strict)
+    .disable_introspection()
+    .finish()
+  {
+    Ok(schema) => Ok(schema),
+    Err(e) => Err(ValidationError::new(e.to_string())),
+  }
+}
+
+pub async fn validate_operation(schema: &Schema, query: &str) -> Vec<Cause<String>> {
+  schema
+    .execute(query)
+    .await
+    .errors
+    .iter()
+    .map(|err| {
+      Cause::new(err.message.clone()).trace(
+        err
+          .locations
+          .iter()
+          .map(|loc| format!("line {} column {}", loc.line, loc.column))
+          .collect(),
+      )
+    })
+    .collect()
+}
+
+pub async fn validate_operations(blueprint: &Blueprint, operations: Vec<String>) -> Valid<(), String> {
```

Comment:
Anything inside of `blueprint` shouldn't be dependent on any kind of IO.
`ValidateOperation` should take in all the operations as string, parse and validate them.

IO Operations should happen within the CLI.